### PR TITLE
The birthday calendar can only hold VEVENT

### PR DESCRIPTION
**Pull request by DeepDiver1975**
refs https://github.com/owncloud/tasks/issues/338

### CODE REVIEW

**Comment by raimund-schluessler** [comment]
Hm, after applying this patch I still get this response for the ```birthday-calendar```

```
 <d:response>
  <d:href>/owncloud/remote.php/dav/calendars/admin/contact_birthdays/</d:href>
  <d:propstat>
   <d:prop>
    <d:displayname>Contact birthdays</d:displayname>
    <cal:calendar-description/>
    <cal:calendar-timezone/>
    <x1:calendar-order xmlns:x1="http://apple.com/ns/ical/">0</x1:calendar-order>
    <x1:calendar-color xmlns:x1="http://apple.com/ns/ical/">#FFFFCA</x1:calendar-color>
    <cal:supported-calendar-component-set>
     <cal:comp name="VEVENT"/>
     <cal:comp name="VTODO"/>
    </cal:supported-calendar-component-set>
    <d:acl>
     <d:ace>
      <d:principal>
       <d:href>/owncloud/remote.php/dav/principals/users/admin/</d:href>
      </d:principal>
      <d:grant>
       <d:privilege>
        <d:read/>
       </d:privilege>
      </d:grant>
      <d:protected/>
     </d:ace>
    </d:acl>
    <d:owner>
     <d:href>/owncloud/remote.php/dav/principals/users/admin/</d:href>
    </d:owner>
    <oc:invite/>
   </d:prop>
   <d:status>HTTP/1.1 200 OK</d:status>
  </d:propstat>
  <d:propstat>
   <d:prop>
    <oc:calendar-enabled/>
   </d:prop>
   <d:status>HTTP/1.1 404 Not Found</d:status>
  </d:propstat>
 </d:response>
 ```

Since there is still a VTODO in it, the Tasks app does still show the calendar.

**Comment by raimund-schluessler** [comment]
I also tried ```php occ dav:sync-birthday-calendar``` with no success.

**Comment by DeepDiver1975** [comment]
I will add a repair step to adjust this for existing birthday calendars

**Comment by raimund-schluessler** [comment]
Great, thanks!
:+1:

**Comment by PVince81** [comment]
@georgehrke second review ?

File: apps/dav/lib/CalDAV/BirthdayService.php [hunk 1]
```diff
@@ -108,6 +108,7 @@ public function ensureCalendarExists($principal) {
 		$this->calDavBackEnd->createCalendar($principal, self::BIRTHDAY_CALENDAR_URI, [
 			'{DAV:}displayname' => 'Contact birthdays',
 			'{http://apple.com/ns/ical/}calendar-color' => '#FFFFCA',
+			'components'   => 'VEVENT,VTODO',
```
Comment (georgehrke) [inline apps/dav/lib/CalDAV/BirthdayService.php]: Why is this 'VEVENT,VTODO` and not just `VTODO`?

Comment (DeepDiver1975) [inline apps/dav/lib/CalDAV/BirthdayService.php]: THX for the hint

**Comment by DeepDiver1975** [comment]
jenkins is green on this one @PVince81 @butonic merge? THX

**Comment by butonic** [comment]
👍

**Comment by thommierother** [comment]
This issue still seems to be present on 9.1.4 with a calendar created in 9.1.3 See also https://github.com/owncloud/tasks/issues/378. It makes the adding of new tasks unsuccessful, at least for users who are not able to de-select "birthdays" and select the real calendar before entering a new task (>> bad navigation in UI).

**Pull request merged**